### ISSUE TITLE
Implement Loss Functions

### ISSUE THREAD

**Issue by JensOstertag** [JensOstertag/ConvolutionalNeuralNetwork#7]
- [x] Mean Squared Error
- [x] Squared Error
- [x] Binary Cross Entropy
